src/palace/manager/util/http.py

```diff
@@ -162,35 +216,37 @@ def set_quick_failure_settings(cls) -> None:
         cls.DEFAULT_REQUEST_TIMEOUT = 5
 
     @classmethod
-    def get_with_timeout(cls, url: str, *args: Any, **kwargs: Any) -> Response:
+    def get_with_timeout(cls, url: str, **kwargs: Unpack[GetRequestKwargs]) -> Response:
         """Make a GET request with timeout handling."""
-        return cls.request_with_timeout("GET", url, *args, **kwargs)
+        return cls.request_with_timeout("GET", url, **kwargs)
 
     @classmethod
     def post_with_timeout(
-        cls, url: str, payload: str | Mapping[str, Any], *args: Any, **kwargs: Any
+        cls,
+        url: str,
+        **kwargs: Unpack[RequestKwargs],
```

Comment:
Functional change here, I dropped the `payload` arg / kwarg, and instead use the `data` kwarg. This brings our implementation into line with requests. I followed this change through everywhere we call this method.